### PR TITLE
improve source code of module loader, api document, eslint etc

### PR DESCRIPTION
**Pull request by chenkaiC4**
- fix eslint error at `loader.base.js` and `loader.volume.js`
- add more api document with `jsDoc Style` at `loader.base.js` and `loader.volume.js`
- improve promise resolve logic

### CODE REVIEW

File: src/loaders/loaders.base.js [hunk 1]
```diff
@@ -101,50 +110,73 @@ export default class LoadersBase {
...
+  /**
+   * parse the data loaded
+   * SHOULD BE implementd by detail loader.
+   * @param {object} response - loaded data.
+   * @return {promise} promise.
+   */
   parse(response) {
     return new Promise((resolve, reject) => {
-      resolve(null);
+      resolve(response);
```
Comment (chenkaiC4) [inline src/loaders/loaders.base.js]: As a interface just return the origin respone.

File: src/loaders/loaders.base.js [hunk 2]
```diff
@@ -101,50 +110,73 @@ export default class LoadersBase {
...
+   * @return {promise} promise.
+   */
   loadSequence(url) {
     return this.fetch(url)
       .then((rawdata) => {
         return this.parse(rawdata);
       })
       .then((data) => {
         this._data.push(data);
+        return data;
```
Comment (chenkaiC4) [inline src/loaders/loaders.base.js]: Return the data, so we can just get the data by:

``` js
loadSequence(url).then((data) => {
  // do with data
})
```

And with `promise.all`, we can get a array filled with data loaded by single resource url.

**Comment by chenkaiC4** [comment]
i don't know why 12 commits list here, this PR shoud just include 3 commits. The first 9 commits have be merged yesterday. 😢

**Comment by NicolasRannou** [comment]
Don't worry about that - when merging I'll squash them all into 1 -

**Comment by NicolasRannou** [comment]
LGTM

**Pull request merged**